pkg/hostagent/snats.go

```diff
@@ -1006,10 +1006,14 @@ func (agent *HostAgent) getMatchingSnatPolicy(obj interface{}) (snatPolicyNames
 				if res == POD {
 					if len(item.Spec.SnatIp) == 0 {
 						var services []*v1.Service
-						cache.ListAllByNamespace(agent.serviceInformer.GetIndexer(), namespace, labels.Everything(),
-							func(servobj interface{}) {
-								services = append(services, servobj.(*v1.Service))
-							})
+						for key, value := range label {
+							singleLabel := make(map[string]string)
```

Comment:
the label in L976 is already a map[string][string], why do we need to create another map[string[string] with the same key, value pairs?

We can instead have selector := labels.SelectorFromSet(labels.Set(snatpolicy.Spec.Selector.Labels)) and use this selector while listing the services

Reply:
@tanyatukade , label variable has multiple labels in it. we we select services using label variable, it will try to get the services which has all those labels in it. But, we need to include the services which has which matches with at least one of the pod labels. Thats why I'm getting labels one by one and adding the service that matches that label.

Reply:
We can have a similar func (https://github.com/noironetworks/aci-containers/blob/0791748e9d0eae3e7f67df67afa93d7c45bb74a0/pkg/controller/snats.go#L169)  in hostagent that will get all the serives using the selector. Can we try this?

Reply:
Since we have access to the pod labels, we can pass that as the selector argument

Reply:
Since SelectorFromSet selects all the labels from the map (i.e the entire set), we wont be able to use it to match individual labels. We should check if the library has any other funcs that selects individual labels instead of the entire set, that way we can avoid the for loop.

Reply:
@tanyatukade , I tried multiple library functions, but couldn't find any function that gets/lists the services that matches one of the given labels. All of them are listing the services that has all the given labels.

Reply:
Thanks for checking, can we add a function that can do this in our utils package? And call that func here?

Reply:
Addressed